src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/AnalyzerTest`1.cs

```diff
@@ -558,16 +560,43 @@ private void VerifyDiagnosticResults(IEnumerable<(Project project, Diagnostic di
                 else
                 {
                     VerifyDiagnosticLocation(analyzers, actual.diagnostic, expected, actual.diagnostic.Location, expected.Spans[0], verifier);
+                    int[] unnecessaryIndices = { };
+                    if (actual.diagnostic.Properties.TryGetValue(WellKnownDiagnosticTags.Unnecessary, out var encodedUnnecessaryLocations))
+                    {
+                        verifier.True(actual.diagnostic.Descriptor.CustomTags.Contains(WellKnownDiagnosticTags.Unnecessary), "Diagnostic reported extended unnecessary locations, but the descriptor is not marked as unnecessary code.");
+                        var match = EncodedIndicesSyntax.Match(encodedUnnecessaryLocations);
+                        verifier.True(match.Success, $"Expected encoded unnecessary locations to be a valid JSON array of non-negative integers: {encodedUnnecessaryLocations}");
+                        unnecessaryIndices = match.Groups["Index"].Captures.OfType<Capture>().Select(capture => int.Parse(capture.Value)).ToArray();
+                        verifier.NotEmpty(nameof(unnecessaryIndices), unnecessaryIndices);
```

Comment:
>verifier.NotEmpty(nameof(unnecessaryIndices), unnecessaryIndices); [](http://example.com/codeflow?start=24&length=66)

nit: not sure which I like better, just wanted to point out that the regex could have been written to not accept an empty number of indices

Reply:
➡️ I felt it would be harder to understand why the regex failed